LegworkSocial/swagger-tools
# LWS-16043 LWS-16107 add default for additionalProperties if not provided

For discussion to decide if forking the origin `swagger-tools` repository and making the changes included here is the best approach (versus modifying the middleware in the service repos).

See:
- [LWS-16043](https://legwork-developers.atlassian.net/browse/LWS-16043) Parent story
- [LWS-16107](https://legwork-developers.atlassian.net/browse/LWS-16107) Swagger fork spike sub-task
- [LWS-16044](https://legwork-developers.atlassian.net/browse/LWS-16044) Sub task associated with a branch that implements the middleware approach.

### Testing
- Choose a service repo (e.g., `message-service`) to test with.
- Modify that service's `package.json` entry for `dependencies > swagger-tools` to
   ```json
   "swagger-tools": "https://github.com/LegworkSocial/swagger-tools#0e92519878ae1721db1f3bd03ea076b94b0e267e",
   ```
- Save and run `yarn` to install the updated `swagger-tools` reference.
- Run some `POST` and `PATCH` endpoints from your service's api. Include properties that are not defined in the `body` parameter's schema. (See the parent story in Jira for an example scenario)
- This should throw a `400` response similar to:
```json
{
    "status": 400,
    "message": "",
    "failedValidation": true,
    "results": {
        "errors": [
            {
                "code": "OBJECT_ADDITIONAL_PROPERTIES",
                "message": "Additional properties not allowed: accountId",
                "path": []
            }
        ],
        "warnings": []
    }
}
```
- Including the `additionalProperties` attribute in the `body` parameter's schema should respect the setting's value (object or boolean).
